### PR TITLE
If the name string is empty, ignore the set-cookie-string entirely

### LINKED ISSUE #1205: Incorrect interpretation of RFC 6265 in SetCookie::validate
[SetCookie::validate](https://github.com/guzzle/guzzle/blob/6692db3cda013e8357330243c33fe1838d0ea6f5/src/Cookie/SetCookie.php#L366) checks if the cookie is valid according to RFC 6265

It includes the following checks:
  1. Names must not be empty, but can be 0
  1. Value must not be empty, but can be 0

[RFC 6265](http://tools.ietf.org/html/rfc6265#page-18)
```
   3.  The (possibly empty) name string consists of the characters up
       to, but not including, the first %x3D ("=") character, and the
       (possibly empty) value string consists of the characters after
       the first %x3D ("=") character.

   4.  Remove any leading or trailing WSP characters from the name
       string and the value string.

   5.  If the name string is empty, ignore the set-cookie-string
       entirely.

```

It only says to ignore the cookie, not to do something drastic like throwing exception, when name string is empty and says nothing about empty value.

### PR DESCRIPTION
**Pull request by halabuda**
really nice library guys!
fixes #1205

### CODE REVIEW

**Comment by mtdowling** [comment]
Sweet! Can you add a test case for this?

File: src/Cookie/CookieJar.php [hunk 1]
```diff
@@ -148,6 +148,11 @@ function (SetCookie $cookie) {
 
     public function setCookie(SetCookie $cookie)
     {
+        //If the name string is empty, ignore the set-cookie-string entirely
+        if(empty($cookie->getName())){
```
Comment (GrahamCampbell) [inline src/Cookie/CookieJar.php]: cs

File: src/Cookie/CookieJar.php [hunk 2]
```diff
@@ -148,6 +148,11 @@ function (SetCookie $cookie) {
 
     public function setCookie(SetCookie $cookie)
     {
+        //If the name string is empty, ignore the set-cookie-string entirely
```
Comment (GrahamCampbell) [inline src/Cookie/CookieJar.php]: cs

**Comment by halabuda** [comment]
@mtdowling sure thing. working on it now.

**Comment by georaldc** [comment]
I think this should follow the "Names must not be empty, but can be 0" rule too, or move the empty name check condition in the validate function outside

**Comment by stefanruijsenaars** [comment]
Added a PR in https://github.com/guzzle/guzzle/pull/1245

**Comment by mtdowling** [comment]
Addressed by #1245

**Pull request closed without merging**